features/admin.validation.v1/components/password-expiry-rule-list.tsx

```diff
@@ -0,0 +1,559 @@
+/**
+ * Copyright (c) 2024, WSO2 LLC. (https://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SelectChangeEvent } from "@mui/material/Select";
+import Alert from "@oxygen-ui/react/Alert";
+import Button from "@oxygen-ui/react/Button";
+import Checkbox from "@oxygen-ui/react/Checkbox";
+import Chip from "@oxygen-ui/react/Chip";
+import Grid from "@oxygen-ui/react/Grid/Grid";
+import IconButton from "@oxygen-ui/react/IconButton";
+import List from "@oxygen-ui/react/List";
+import ListItem from "@oxygen-ui/react/ListItem";
+import ListItemText from "@oxygen-ui/react/ListItemText";
+import MenuItem from "@oxygen-ui/react/MenuItem";
+import Select from "@oxygen-ui/react/Select";
+import TextField from "@oxygen-ui/react/TextField";
+import { ChevronDownIcon, ChevronUpIcon, PlusIcon, TrashIcon } from "@oxygen-ui/react-icons";
+import { userstoresConfig } from "@wso2is/admin.extensions.v1";
+import { GroupsInterface } from "@wso2is/admin.groups.v1";
+import {
+    GovernanceConnectorConstants
+} from "@wso2is/admin.server-configurations.v1/constants/governance-connector-constants";
+import { RolesInterface } from "@wso2is/core/models";
+import React, { FunctionComponent, ReactElement, useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { PasswordExpiryRule, PasswordExpiryRuleAttribute, PasswordExpiryRuleOperator } from "../models";
+
+interface PasswordExpiryRuleListProps {
+    componentId: string;
+    ruleList: PasswordExpiryRule[];
+    isPasswordExpiryEnabled: boolean;
+    isSkipFallbackEnabled: boolean;
+    defaultPasswordExpiryTime: number;
+    rolesList: RolesInterface[];
+    groupsList: GroupsInterface[];
+    isReadOnly: boolean;
+    onSkipFallbackChange: (value: boolean) => void;
+    onDefaultPasswordExpiryTimeChange: (value: number) => void;
+    onRuleChange: (rules: PasswordExpiryRule[]) => void;
+    onRuleError: (hasErrors: boolean) => void;
+}
+
+type Resource = RolesInterface | GroupsInterface;
+
+export const PasswordExpiryRuleList: FunctionComponent<PasswordExpiryRuleListProps> = (
+    props: PasswordExpiryRuleListProps
+) => {
+    const {
```

Comment:
This props pattern usage is deprecated and going forward we should get these params from the function itself.  Refer: https://github.com/wso2/identity-apps/blob/3c1a7c3d215c8344703548eda075ba2c10504c2c/features/admin.template-core.v1/components/resource-create-wizard.tsx#L122-L140

Reply:
Addressed in https://github.com/wso2/identity-apps/pull/6862.